### DISCUSSION TITLE
Alias pattern

### DISCUSSION THREAD

**Discussion by Hanse00** [Homebrew/discussions#4516] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.18-85-g16b2a14
ORIGIN: https://github.com/Homebrew/brew
HEAD: 16b2a14c9408140b04828462b660d30056421fdb
Last commit: 7 hours ago
Core tap JSON: 18 May 21:22 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I recently found myself wanting to install node via homebrew. To my surprise `brew search node` returned versioned formulae for versions 10, 14, 16, and 18, but not 20. After a bit of additional digging, I realized that the `brew` (no version) formula is indeed version 20.

Looking through the repo, I then realized there's a `node@20` _alias_ which points to the `node` formula.

This is in contrast to the setup for python, which each formula is explicitly numbered, with `python` and `python@3` aliases pointing to the actual numbered formula.

That brings me to my question: Is there a standard pattern for handling formulae with multiple versions? I would personally be in favor of formulae being version specific, and having generic version aliases pointing at them (like python) rather than generic formulae with specific version pointers (like node).

**Comment by SMillerDev** [comment]
Yes, by default the latest version will not include the version number but will have an alias with the exception of python and OpenSSL since they are too difficult to migrate otherwise.
